app/models/member_with_authentication.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+class MemberWithAuthentication
```

Comment:
indeed. Really, this could/should in fact be a service class (despite that fact that it behaves in some regards, like a model).